jsettlers.logic/src/jsettlers/ai/construction/BestMilitaryConstructionPositionFinder.java

```diff
@@ -173,7 +190,10 @@ public ShortPoint2D findBestConstructionPosition(AiStatistics aiStatistics, Abst
 				aiStatistics.getNearestResourcePointInDefaultPartitionFor(referencePoint, EResourceType.GOLDORE, Integer.MAX_VALUE) != null) {
 			importantResources.add(ImportantResource.GOLD);
 		}
-		if (importantResources.size() == 0) {
+		// 50 : 50 chance to spread the land or to go for more resources
+		if (importantResources.size() == 0 && RandomSingleton.getBoolean() == true) {
```

Comment:
We cannot use the RandomSingleton in the AI. This is because the AI is only calculated in the game of one player. Therefore this player would have a differing use of RandomSingleton from the others (if bots are used in multiplayer games). 
However, we also cannot use Math.random(), as this would make the AIs behaviour interdeministic and therefore unreproduceable which would make debugging a pain. 

My proposal: Add an AI random generator only used by the AI and initialized with the same seed as the RandomSingleton (or the first random number of the random singleton). The important thing is that this seed is deterministic but not fixed.